auth_mellon_config.c

```diff
@@ -855,6 +855,52 @@ static const char *am_set_do_not_verify_logout_signature(cmd_parms *cmd,
 #endif
 }
 
+/* This function handles the MellonMergeEnvVars configuration directive, 
+ * it sets merge_env_vars to nonempty separator (default semicolon),
+ * or empty string to denote no merging.
+ *
+ * Parameters:
+ *  cmd_parms *cmd       The command structure for this configuration
+ *                       directive.
+ *  void *struct_ptr     Pointer to the current directory configuration.
+ *                       NULL if we are not in a directory configuration.
+ *  const char *flag     On/Off flag
+ *  const char *sep      Optional separator, should be only present with On
+ *
+ * Returns:
+ *  NULL on success or an error string on failure.
+ */
+static const char *am_set_merge_env_vars(cmd_parms *cmd,
+                                          void *struct_ptr,
+                                          const char *flag,
+                                          const char *sep)
+{
+    am_dir_cfg_rec *d = (am_dir_cfg_rec *)struct_ptr;
+    apr_pool_t *p= cmd->pool;
+    if (strcasecmp(flag, "on") == 0) {
+        if (sep && *sep) {
```

Comment:
If `AP_INIT_TAKE12` will not include the second argument if it is an empty string, why is the check for `*sep` required?